crates/kubelet/src/state/common/mod.rs

```diff
@@ -50,21 +52,21 @@ pub trait GenericProviderState: 'static + Send + Sync {
 /// Exposes pod state in a way that can be consumed by
 /// the generic states.
 #[async_trait::async_trait]
-pub trait GenericPodState: 'static + Send + Sync {
+pub trait GenericPodState: ResourceState<Manifest = Pod, Status = PodStatus> {
     /// Stores the pod module binaries for future execution. Typically your
     /// implementation can just move the modules map into a member field.
-    fn set_modules(&mut self, modules: HashMap<String, Vec<u8>>);
+    async fn set_modules(&mut self, modules: HashMap<String, Vec<u8>>);
```

Comment:
Making these async feels like it should be a separate design discussion and a separate commit, not rolled into the container state changes.

Reply:
I think the reason this triggered the discussion is that the container state machines need access to these data structures concurrently. I think there is an argument to be made for moving the image pull into the container state machine so that they do not need this shared reference. 

I think we should establish a policy of all-async, but we can discuss that later.